src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,97 @@
+import styled from '@emotion/styled';
+import { executeUpdate } from 'actions/metadataActions';
+import * as metadataActions from 'actions/metadataActions';
+import FooterUpdateButton from 'Footer/FooterUpdateButton';
+import {
+  getReleaseURL,
+  getVersionTooltipMessage,
+  hasUpdateReady,
+  showUpdateToast,
+} from 'Footer/FooterUtils';
+import FooterVersion from 'Footer/FooterVersion';
+import React, { useCallback, useEffect, useRef } from 'react';
+import OverlayTrigger from 'react-bootstrap/lib/OverlayTrigger';
+import Tooltip from 'react-bootstrap/lib/Tooltip';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  getMetadataCurrentVersion,
+  getMetadataIsUpdating,
+  getMetadataNewVersion,
+} from 'selectors/metadataSelectors';
+
+const FooterGroup = styled.span`
+  & + & {
+    margin-left: 0.35rem;
+  }
+`;
+
+interface IFooterProps extends React.ComponentPropsWithoutRef<'footer'> {}
+
+const Footer: React.FC<IFooterProps> = (props: IFooterProps) => {
+  const dispatch = useDispatch();
+
+  const currentVersion: string = useSelector(getMetadataCurrentVersion);
+  const newVersion: string | null = useSelector(getMetadataNewVersion);
+  const isUpdating: boolean = useSelector(getMetadataIsUpdating);
+
+  const isToastVisible: React.MutableRefObject<boolean> = useRef<boolean>(
+    false
+  );
+
+  const tooltipMessage: string = getVersionTooltipMessage(
+    currentVersion,
+    newVersion,
+    isUpdating
+  );
+  const releaseURL: string = getReleaseURL(currentVersion);
+  const isUpdateReady: boolean = hasUpdateReady(currentVersion, newVersion);
+
+  const handleUpdate = useCallback(() => {
+    dispatch(executeUpdate());
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(metadataActions.setInitialVersion());
+    metadataActions.registerUpdateChecker(dispatch);
```

Comment:
Is it possible that we are registering multiple 'setTimeouts' like this?

I think we should guard against that and take the timeout ID and store it somewhere. Then check if it already exists, and either clear the existing one and set a new one, or just not set any new timeout.

Reply:
About the request mocks not being manually added to all tests that need it, I remember us talking about having a more granular control over which and how many requests are executed, and extracting some requests to a separate helper function kind of goes against that. We could re-address that, though.

As for the timeouts, I would usually store the timeout, but in this case, does it really make a difference? The `Footer` component is rendered at the root of the app, and the timeout is created when the `Footer` component is mounted. The only situation where the timeout would be created again would be if the whole app would be unmounted and mounted again. Is that possible though?

Reply:
However, if our logic ever changes and we want to change where the `Footer` component is rendered, we will 99% sure forget about the timeout thing, so it would probably be safer to store it, to not have to deal with it later

Reply:
True, the footer only mounts once and that effect would only fire again if 'dispatch' changes in the props right.

I guess the only remaining argument if is in some future we decide to change something in the footer or that effect which could potentially cause re-rendering, then we might forget that the effect creates a timeout and introduce an unintended side effect.